simulators/discovery/version4/discv4.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/ethereum/hive/simulators/common"
+
+	"github.com/ethereum/hive/simulators/common/providers/hive"
+)
+
+var (
+	host     common.TestSuiteHost
+	suiteID  common.TestSuiteID
+	testID   common.TestID
+	clientID string
+)
+
+func main() {
+	host = hive.New()
+
+	// get clients
+	// loop over individual clients
+	// get node
+	// start test
+
+	clients, err := host.GetClientTypes()
+	if err != nil {
+		fatalf("could not get clients: %v", err)
+	}
+
+	logFile := os.Getenv("HIVE_SIMLOG")
+
+	for _, client := range clients {
+		env := map[string]string{
+			"CLIENT":              client,
+			"HIVE_NETWORK_ID":     "19763",
+			"HIVE_CHAIN_ID":       "19763",
+			"HIVE_FORK_HOMESTEAD": "0",
+			"HIVE_FORK_TANGERINE": "0",
+			"HIVE_FORK_SPURIOUS":  "0",
+			"HIVE_FORK_BYZANTIUM": "0",
+			"HIVE_LOGLEVEL":       "5",
+		}
+
+		var err error
+		suiteID, err = host.StartTestSuite("discv4", "discovery v4 test", logFile)
+		if err != nil {
+			fatalf("could not start test suite: %v", err)
+		}
+
+		testID, err = host.StartTest(suiteID, "discovery v4", "This test suite tests a "+
+			"client's conformance to the discovery v4 protocol.")
+		if err != nil {
+			fatalf("could not start test: %v", err)
+		}
+		// get container ID and bridge IP of the client
+		clientID, _, _, err := host.GetNode(suiteID, testID, env, nil) // TODO can pass nil for files?
+		if err != nil {
+			endTest(&common.TestResult{
+				Pass:    false,
+				Details: err.Error(),
+			})
+			fatalf("error getting node: %v", err)
+		}
+		// get enode ID from client
+		enode, err := host.GetClientEnode(suiteID, testID, clientID)
+		if err != nil {
+			endTest(&common.TestResult{
+				Pass:    false,
+				Details: err.Error(),
+			})
+			fatalf("error getting node peer-to-peer endpoint: %v", err)
+		}
+		// create a separate network to be able to send the client traffic from two separate IP addrs
+		networkID, err := host.CreateNetwork(suiteID, "network1")
+		if err != nil {
+			endTest(&common.TestResult{
+				Pass:    false,
+				Details: err.Error(),
+			})
+			fatalf("could not create network: %v", err)
+		}
+		// connect both simulation and client to this network
+		err = host.ConnectContainer(suiteID, networkID, clientID)
+		if err != nil {
+			endTest(&common.TestResult{
+				Pass:    false,
+				Details: err.Error(),
+			})
+			fatalf("could not connect container to network: %v", err)
+		}
+		err = host.ConnectContainer(suiteID, networkID, "simulation")
+		if err != nil {
+			endTest(&common.TestResult{
+				Pass:    false,
+				Details: err.Error(),
+			})
+			fatalf("could not connect container to network: %v", err)
+		}
+		// get bridge network ID
+		bridgeID, err := host.GetNetworkID(suiteID, "bridge")
```

Comment:
This part I don't like. It's a lot like the situation we had before with the simulation container. I think we should solve this by having a special, `"default"` network ID, which is recognized by the API and translated into the actual network ID of the default network.